src/main/java/net/dv8tion/jda/api/entities/templates/TemplateChannel.java

```diff
@@ -0,0 +1,337 @@
+     * The more recent a channel was created, the lower it is in the hierarchy.
+     *
+     * @return The true, Discord stored, position of the {@link net.dv8tion.jda.api.entities.templates.TemplateChannel TemplateChannel}.
+     */
+    public int getPositionRaw()
+    {
+        return this.rawPosition;
+    }
+
+    /**
+     * Parent Category id of this TemplateChannel. Channels don't need to have a parent Category.
+     * <br>Note that a Category channel will always return {@code -1} for this method
+     * as nested categories are not supported.
+     *
+     * @return The id of the parent Category or {@code -1} if the channel doesn't have a parent Category
+     */
+    public long getParentId()
+    {
+        return this.parentId;
+    }
+
+    /**
+     * Whether or not this channel is considered as "NSFW" (Not-Safe-For-Work).
+     * <br>If the {@link net.dv8tion.jda.api.entities.ChannelType ChannelType}
+     * <b>is not {@link net.dv8tion.jda.api.entities.ChannelType#TEXT TEXT}</b>, this returns {@code false}.
+     *
+     * @return Whether this TextChannel is considered NSFW or {@code false} if the channel is not a text channel
+     */
+    public boolean isNSFW()
+    {
+        return this.nsfw;
+    }
+
+    /**
+     * The slowmode set for this TemplateChannel.
+     * <br>If slowmode is set this returns an {@code int} between 1 and {@link net.dv8tion.jda.api.entities.TextChannel#MAX_SLOWMODE TextChannel.MAX_SLOWMODE}.
+     * <br>If not set this returns {@code 0} or if the {@link net.dv8tion.jda.api.entities.ChannelType ChannelType}
+     * <b>is not {@link net.dv8tion.jda.api.entities.ChannelType#TEXT TEXT}</b>, this returns {@code -1}.
+     *
+     * <p>Note bots are unaffected by this.
+     * <br>Having {@link net.dv8tion.jda.api.Permission#MESSAGE_MANAGE MESSAGE_MANAGE} or
+     * {@link net.dv8tion.jda.api.Permission#MANAGE_CHANNEL MANAGE_CHANNEL} permission also
+     * grants immunity to slowmode.
+     *
+     * @return The slowmode for this TextChannel, between 1 and {@link net.dv8tion.jda.api.entities.TextChannel#MAX_SLOWMODE TextChannel.MAX_SLOWMODE}, {@code 0} if no slowmode is set or {@code -1} if the channel is not a text channel
+     */
+    public int getSlowmode()
+    {
+        return this.channelType == ChannelType.TEXT ? this.slowmode : -1;
+    }
+
+    /**
+     * The audio bitrate of the voice audio that is transmitted in this channel. While higher bitrates can be sent to
+     * this channel, it will be scaled down by the client.
+     * <br>If the {@link net.dv8tion.jda.api.entities.ChannelType ChannelType}
+     * <b>is not {@link net.dv8tion.jda.api.entities.ChannelType#VOICE VOICE}</b>, this returns {@code -1}.
+     *
+     * <br>Default and recommended value is 64000
+     *
+     * @return The audio bitrate of this voice channel
+     */
+    public int getBitrate()
+    {
+        return this.channelType == ChannelType.VOICE ? this.bitrate : -1;
```

Comment:
I'd prefer to remove these checks entirely and just return whatever discord gave us.